CONTRIBUTING.md

```diff
@@ -0,0 +1,127 @@
+# CONTRIBUTING
+
+Thank you for considering contributing to the Outrigger CLI!
+
+## Quality Contributions
+
+* Make sure your branch will compile.
+* Make sure your branch passes our static analysis checks.
+* Make sure your branch conforms with go fmt standards.
+* Manually test your changes.
+
+## User Interactions
+
+One of the key goals of this project is to promote a positive developer
+experience. Every interaction should be thought of with the following points:
+
+* Are you providing the user with enough context about what's being asked or being done?
+* Does the user expect to wait? Might the user think the tool stalled?
+* Is there black box business happening that could be made more transparent?
+
+We have a slightly complex logging API to support addressing these concerns.
+(See ./util/logging.go)
+
+Here are a few conventions:
+
+* **Starting a task that could take more than 5 seconds:**
+  * `cmd.out.Spin("Preparing the sauce")`
+* **Use the correct method to log operational results: (Pick one)**
+  * `cmd.out.Success("Sauce is Ready.")`
+  * `cmd.out.Warn("Sauce is burnt on the bottom.")`
+  * `cmd.out.Oops("Discard this sauce and try again.")`
+* **Going to send some contextual notes to the user**:
+  1. `cmd.out.NoSpin()` if currently using the spinner.
+  2. `cmd.out.Status("Sauce exists.")`
+  3. `cmd.out.Note("This is a list item.")`
+  4. `cmd.out.Details("The ingredients of the sauce include tomato, salt, black pepper, garlic...")`
+* **Command has executed and is successful. We do not want a notification:**
+  ```
+  cmd.out.Success("Enjoy your dinner.")
+  return cmd.Success("")
```

Comment:
`cmd.out.Success()` and `cmd.Success()` is confusing. Can they be consolidated?, or can we rename one?

Reply:
We may also want a `SilentSuccess()` so we dont need to send in `""`?

Reply:
yeah, or `SuccessWithNotification()`

I agree the two Success() statements is confusing. I originally had it as `Complete()`, but complete only made sense in the context of the spinner. Unfortunately this "success" is often the equivalent of an info log. I will look at the API overall and see what can be rejiggered. Now that I know I'm on the right track I can make deeper cuts.